airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/auth/AuthButton.test.tsx

```diff
@@ -0,0 +1,148 @@
+import { screen, render } from "@testing-library/react";
+
+import { TestWrapper } from "utils/testutils";
+import { useServiceForm } from "views/Connector/ServiceForm/serviceFormContext";
+
+import { AuthButton } from "./AuthButton";
+import { useFormikOauthAdapter } from "./useOauthFlowAdapter";
+jest.setTimeout(10000);
+
+jest.mock("formik", () => {
+  const useFormikContext = () => {
+    const values = {};
+    const submitCount = 0;
+    return { values, submitCount };
+  };
+
+  return { useFormikContext };
+});
+
+/**
+ * Mock services to reuse multiple times with different values:
+ * 1. mock the service
+ * 2. cast the type as shown below.  `as unknown` is required to get the typing to play nice as there isn't crossover
+ *    Partial is optional, but required if you do not want to mock the entire object
+ *
+ * Then, can implement in tests using useWhateverServiceYouMocked.mockImplementationOnce or useWhateverServiceYouMocked.mockImplementation
```

Comment:
Some funky workarounds to be able to use `mockImplementationOnce`, but I wanted to change the mocked return values for each test and I _think_ this is the cleanest way to do it (based on [this](https://stackoverflow.com/questions/52457575/jest-typescript-property-mock-does-not-exist-on-type) and similar threads)